test/truefi2/TrueCreditAgency.test.ts

```diff
@@ -150,6 +152,21 @@ describe('TrueCreditAgency', () => {
       await expect(creditAgency.connect(borrower).borrow(tusdPool.address, 1000))
         .to.be.revertedWith('TrueCreditAgency: The pool is not whitelisted for borrowing')
     })
+
+    it('correctly handles the case when credit score is changing', async () => {
+      await creditAgency.connect(borrower).borrow(tusdPool.address, 1000)
+      expect(await creditAgency.creditScore(tusdPool.address, borrower.address)).to.equal(255)
+      expect((await creditAgency.buckets(tusdPool.address, 255)).totalBorrowed).to.equal(1000)
+      expect((await creditAgency.buckets(tusdPool.address, 255)).borrowersCount).to.equal(1)
+
+      await creditOracle.setScore(borrower.address, 200)
+      await creditAgency.connect(borrower).borrow(tusdPool.address, 1000)
+      expect(await creditAgency.creditScore(tusdPool.address, borrower.address)).to.equal(200)
```

Comment:
Should `creditScore` be "per pool"?

Reply:
It's not perfect, but we have to update credit score separately. We probably want to change this later and iterate over all pools during updates, but for now it's a workaround.

Reply:
Maybe `mapping(ITrueFiPool2 => mapping(address => uint8)) public creditScore;` can be just `mapping(address => uint8) public creditScore;`? And we we will just always omit the pool part?
Instead of always typing `creditScore[pool][user]`, we'll just type `creditScore[user]`? Or will it cause severe problems with rebucketing?

Reply:
We don't know if score was updated for the pool so we don't know which bucket we should look for

Reply:
Isn't score the same for every pool?